### DISCUSSION TITLE
Filtering apptable backed data

### DISCUSSION THREAD

**Discussion by paperstack** [anvilistas/tabulator#112] [Q&A]
Hello, 

I was hoping to use column header filters to filter on data that's provided via apptables, setup as follows:
```
self.tabulator.options = {
        "app_table": app_tables.my_data,
        "data_loader": False,
        "loading_indicator": True,
        "layout_columns_on_new_data": False,
        "layout": "fitDataFill"
    }
```
I then create the filter via standard  column definitions ("status" is a column in `my_data`):
```
self.tabulator.columns = [
    {"title":"Status", "field":"status", "headerFilter":"input"}
]
```

While the filter appears in the column header, typing in text does nothing. Providing the same data via `self.tabulator.data = my_data` does allow filtering to work. 

Is there anything I'm missing in terms of wiring of how to get the column filters to work as query param to the apptable data fetch?

**Comment by s-cork** [comment]
Minimal clone please and then I'm happy to have a look and play around. 

But the short answer is it won't work. 
Tabulator filtering and data table filtering aren't connected. When you're using the data table option I would need to convert the tabulator filtering syntax to anvils querying syntax. 

You'd need a custom input box that on change, possibly with a delay, sets the query. 

The other option would be for me to translate the filtering to queries. But that's something I avoided the first time for a reason.
